server/src/main/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImpl.java

```diff
@@ -955,14 +978,20 @@ private UserVm importVirtualMachineInternal(final UnmanagedInstanceTO unmanagedI
         }
         // Check NICs and supplied networks
         Map<String, Network.IpAddresses> nicIpAddressMap = getNicIpAddresses(unmanagedInstance.getNics(), callerNicIpAddressMap);
-        Map<String, Long> allNicNetworkMap = getUnmanagedNicNetworkMap(unmanagedInstance.getName(), unmanagedInstance.getNics(), nicNetworkMap, nicIpAddressMap, zone, hostName, owner);
+        Map<String, Long> allNicNetworkMap = getUnmanagedNicNetworkMap(unmanagedInstance.getName(), unmanagedInstance.getNics(), nicNetworkMap, nicIpAddressMap, zone, hostName, owner, host.getHypervisorType());
         if (!CollectionUtils.isEmpty(unmanagedInstance.getNics())) {
             allDetails.put(VmDetailConstants.NIC_ADAPTER, unmanagedInstance.getNics().get(0).getAdapterType());
         }
+
+        if (StringUtils.isNotEmpty(unmanagedInstance.getVncPassword())) {
```

Comment:
should there be a hypervisor check?

Reply:
I don't think its needed as for missing VNC password, we don't execute it at all and code remains simple?

(My thinking was that hypervisors plugins that need to pass back VNC password will set it in UnmanagedInstanceTO, and if it is passed we process it here without worrying about hypervisor type.)